flow360/component/simulation/primitives.py

```diff
@@ -649,13 +649,27 @@ class SnappyBody(Flow360BaseModel):
     body_name: str = pd.Field()
 
 
-class MeshZone(Flow360BaseModel):
+@final
+class SeedpointZone(_VolumeEntityBase):
     """
     Represents a separate zone in the mesh, defined by a point inside it.
+    To be used only with snappyHexMesh.
     """
 
-    point_in_mesh: LengthType.Point
-    name: str
+    private_attribute_entity_type_name: Literal["SeedpointZone"] = pd.Field(
+        "SeedpointZone", frozen=True
+    )
+    type: Literal["SeedpointZone"] = pd.Field("SeedpointZone", frozen=True)
+    point_in_mesh: LengthType.Point = pd.Field(
+        description="Seedpoint for a main fluid zone in snappyHexMesh."
+    )
+    axes: Optional[OrthogonalAxes] = pd.Field(
+        None, description="Principal axes definition when using with PorousMedium"
+    )  # Porous media support
+    axis: Optional[Axis] = pd.Field(None)  # Rotation support
+    # pylint: disable=no-member
+    center: Optional[LengthType.Point] = pd.Field(None, description="")  # Rotation support
```

Comment:
I think these options should be handled by custom constructors:
https://github.com/flexcompute/Flow360/blob/0c6e0a50aa0d79e6cd73b4e91826e025d680dbd3/flow360/component/simulation/primitives.py#L257